eng/versioning/update_changelog.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import sys
+import os
+import logging
+import datetime
+import re
+
+CHANGE_LOG_FILE_NAME = "history.md"
+UNRELEASED_TAG = "(Unreleased)"
+VERSION_REGEX = "\d+\.\d+\.\d+([^0-9\s][^\s:]+)?"
+
+
+def get_version_header(version, isUnreleased):
+    release_date = ""
+    if isUnreleased:
+        release_date = UNRELEASED_TAG
+    else:
+        release_date = datetime.date.today()
+
+    version_header = "## {0} {1}\n".format(version, release_date)
+    return version_header
+
+
+def get_change_log_path(setup_py_location):
```

Comment:
This is yet another place we have special logic to find and parse our changelogs, see https://github.com/Azure/azure-sdk-for-python/issues/8244 where we already have an issue to consolidate that logic. On top of that this code is going to be needed for every other language repo, so I'd like to propose instead we pull out the changelog log for parsing and adding new entries into a set of shared scripts/steps all languages can use and put it into the azure-sdk repo instead.